### ISSUE TITLE
[REQUEST] PVR Recordings

### ISSUE THREAD

**Issue by bebetaz** [OmerTu/GoogleHomeKodi#159]
Hi,

First of all, great work, I have just installed this and I think it will be used everyday for a very long time.

Now, I am using TVHeadend as a PVR back end and I was wondering if it would be possible to include a recording command.
Maybe to start with a direct "hey google, Kodi record channel 10" to start recording immediately, or maybe "hey google, Kodi record channel 10 tomorrow at 8:30pm" for something a bit more complicated. Or, a full blown "hey google, Kodi record The Great Gatsby on channel 7" or "hey google, Kodi record all "Games if Throne" on ABC" to set an automatic recording schedule of single movie or TV series.

Anyhow, again, great work and hopefully will become even greater ;)

Cheers,
bebetaz

**Comment by p0psicles** [comment]
Yeah I have tvheadend setup. So I could experiment a little. But I have been out of this for a while now.

**Comment by bebetaz** [comment]
I'm happy to do some testing if you want, can't help with development but I can follow instructions lol

**Comment by mpember** [comment]
While I don't use this project, I do have my own code to let me control Kodi from my Google AIY gode.

The problem with the current Kodi API is that the PVR controls are limited.  And the documentation is worse.  While it is possible to use the JSON API to find the most recent unwatched episode of a downloaded TV show, the PVR API only lets you download a complete list of the available items.  e.g. My code needed to download the complete list of available PVR channels, then check each one to see if it matches the channel I requested.

**Comment by themusingcogitator** [comment]
My understanding is the next update to Kodi will support Actions on Google. I wonder will PVR support will be added. Of course, each PVR server may require a different command structure. If so, possibly the only current way would be an add-on for each server.

**Comment by mpember** [comment]
> Of course, each PVR server may require a different command structure. If so, possibly the only current way would be an add-on for each server.

In the same way that the GUI and JSON API provides a standard method of accessing the content, it would be the same for any "Actions on Google" API.  There would be a "lowest common denominator" set of functionality.  There should be no need for a different set of code for each PVR backend.

**Comment by bebetaz** [comment]
Hi everyone, just wondering if there is any news on this suggestion.
Cheers.

**Comment by keydon** [comment]
i dont think there is any progress, sorry.
